back/src/domains/convention/adapters/PgConventionRepository.ts

```diff
@@ -51,6 +51,26 @@ export class PgConventionRepository implements ConventionRepository {
     return result.map(({ id }) => id);
   }
 
+  public async deleteOldConventions(updatedBefore: Date) {
+    const result = await this.transaction
+      .with("conventionRenewedFrom", (qb) =>
+        qb
+          .selectFrom("conventions")
+          .select("renewed_from")
+          .where("renewed_from", "is not", null),
+      )
+      .deleteFrom("conventions")
+      .where("updated_at", "<=", updatedBefore)
+      .where("status", "in", ["DEPRECATED", "CANCELLED", "REJECTED"])
+      .where("id", "not in", (eb) =>
+        eb.selectFrom("conventionRenewedFrom").select("renewed_from"),
+      )
+      .returning("id")
+      .execute();
```

Comment:
tu peux peut-être simplifier, and ajouter une clause `.where("renewed_from", "is not", null)` directement à la query principale plutôt que de refaire un `id not in(...)`


```suggestion
      deleteFrom("conventions")
      .where("updated_at", "<=", updatedBefore)
      .where("status", "in", ["DEPRECATED", "CANCELLED", "REJECTED"])
      .where("renewed_from", "is not", null)
      .returning("id")
      .execute();
```